scheds/rust/scx_rustland/src/bpf/main.bpf.c

```diff
@@ -511,7 +519,7 @@ void BPF_STRUCT_OPS(rustland_enqueue, struct task_struct *p, u64 enq_flags)
 	 * potentially slow down the entire system if they are blocked for too
 	 * long (i.e., ksoftirqd/N, rcuop/N, etc.).
 	 */
-	if (is_kthread(p) && p->nr_cpus_allowed == 1) {
+	if (!full_user && is_kthread(p) && p->nr_cpus_allowed == 1) {
```

Comment:
Out of curiosity, how stable is `full_user` mode? This opens up  other non-memory-allocation dependencies and the surface for long tail deadlock events may get a lot larger, I think. I suppose it really depends on how much the userland scheduler interacts with the kernel.

Reply:
I wasn't sure to include this specific condition in `full-user` mode (usually I don't, but without this it wouldn't be "full user"...).

If we don't include this condition `full-user` mode seems to be pretty stable, at least for the tests that I've done. System is less responsive when it's mostly idle, of course, because tasks are not dispatched directly, but that is expected. And when the system is busy even the regular mode becomes "full-user" (no idle cpus => no direct dispatch => full-user), so no difference in this case.

Thinking more about this, we should probably avoid sending per-cpu kthread to the user-space scheduler and consider `full-user` just as a way to disable direct dispatch from BPF. In this way everything would be stable and good enough for testing this particular behavior.

Reply:
Alright, updated `full-user` to exclude single-CPU kthread and make it more stable / reliable. I like it better.